src/http.rs

```diff
@@ -75,39 +62,21 @@ pub async fn do_http_request(
         add_metric_entry!(err_host_not_allowed, rpc_host.clone(), 1);
         return Err(ValidationError::HostNotAllowed(rpc_host.0).into());
     }
-    if !is_authorized(&caller, Auth::FreeRpc) {
```

Comment:
The test runner previously used the `Auth::FreeRpc` permission to send requests without payment (which is nontrivial or potentially not possible in `ic-state-machine-tests`). 

Perhaps we could include an init/upgrade arg to deactivate cycles payments for local testing? This would have the secondary benefit of giving newer developers a faster way to start using the canister if they're unfamiliar with how to call canister methods with cycles (or just want to try the canister through Candid UI).

Reply:
> Is that something that would be possible with pocket-ic?

Yes, I believe so. We could potentially rewrite the relevant tests to use pocket-ic, although this might be a large undertaking since almost all of the tests make at least one RPC call. 

> So if possible, I would much prefer that we test the production system

Definitely. The compromise right now is to test the production system using the "Motoko E2E tests" on both the local replica and mainnet, while the state machine tests cover application logic aside with the exception of cycles payments.

Reply:
> Yes, I believe so. We could potentially rewrite the relevant tests to use pocket-ic, although this might be a large undertaking since almost all of the tests make at least one RPC call.

Yes this we can do later, other ticket/PR.

See above comment for possible solution.